common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -225,76 +246,137 @@ public String encrypt(final String clearText)
+        final String methodName = ":decrypt";
+        Logger.verbose(TAG + methodName, "Starting decryption");
 
         if (StringExtensions.isNullOrBlank(encryptedBlob)) {
             throw new IllegalArgumentException("Input is empty or null");
         }
 
-        int encodeVersionLength = encryptedBlob.charAt(0) - 'a';
-        validateEncodeVersion(encryptedBlob, encodeVersionLength);
+        if (getEncryptionType(encryptedBlob) == EncryptionType.UNENCRYPTED) {
+            Logger.warn(TAG + methodName, "This string is not encrypted. Finished decryption.");
+            return encryptedBlob;
+        }
 
-        final byte[] bytes = Base64.decode(
-                encryptedBlob.substring(1 + encodeVersionLength),
-                Base64.DEFAULT
-        );
+        final String packageName = getPackageName();
+        final List<KeyType> keysForDecryptionType = getKeysForDecryptionType(encryptedBlob, packageName);
+
+        final byte[] bytes = getByteArrayFromEncryptedBlob(encryptedBlob);
+        for (final KeyType keyType : keysForDecryptionType) {
+            try {
+                final SecretKey secretKey = loadSecretKey(keyType);
+                if (secretKey == null) {
+                    continue;
+                }
+
+                String result = decryptWithSecretKey(bytes, secretKey);
+                Logger.verbose(TAG + methodName, "Finished decryption.");
+                return result;
+            } catch (GeneralSecurityException | IOException e) {
+                Logger.error(
+                        TAG + methodName,
+                        "Failed to decrypt with KeyType: "
+                                + keyType.toString(),
+                        e
+                );
+            }
+        }
+
+        Logger.verbose(
+                TAG + methodName,
+                "Tried all decryption keys and decryption still fails. Throw an exception.");
+
+        throw new GeneralSecurityException(ErrorStrings.DECRYPTION_FAILED);
+    }
 
-        final String packageName = mContext.getPackageName();
+    /**
+     * Determine type of encryption performed on the given data blob.
+     * NOTE :If it cannot verify the keyVersion, it will assume that this data is not encrypted.
+     * */
+    public EncryptionType getEncryptionType(@NonNull final String data) throws IOException {
+        final String methodName = ":getEncryptionType";
 
+        final byte[] bytes;
         try {
-            return decryptWithPackageName(bytes, packageName);
-        } catch (GeneralSecurityException | IOException e) {
-            Logger.error(
-                    TAG + methodName,
-                    "Failed to decrypt with package name: "
-                            + packageName,
-                    e
+            bytes = getByteArrayFromEncryptedBlob(data);
+        } catch (IllegalArgumentException e) {
+            Logger.error(TAG + methodName, "This data is not an encrypted blob.", e);
```

Comment:
Let's not log this as error instead have a verbose log as we use this method to determine the type , and for unencrypted type it shows an error which will be a false alarm